HEC.FDA.Model/compute/ImpactAreaScenarioSimulation.cs

```diff
@@ -93,10 +94,9 @@ public ImpactAreaScenarioResults Compute(IProvideRandomNumbers randomProvider, C
         public ImpactAreaScenarioResults Compute(IProvideRandomNumbers randomProvider, ConvergenceCriteria convergenceCriteria, CancellationToken cancellationToken,
             bool computeDefaultThreshold = true, bool giveMeADamageFrequency = false, bool computeIsDeterministic = false)
         {
-            //Validate();
-            if (!CanCompute(convergenceCriteria, randomProvider))
+            if (!CanCompute(convergenceCriteria))
             {
-                _ImpactAreaScenarioResults = new ImpactAreaScenarioResults(_ImpactAreaID, true);
+                _ImpactAreaScenarioResults = new ImpactAreaScenarioResults(_ImpactAreaID, true); //I would like to just return regular Null here but I'm unsure who is relying on this behavior. BBB
                 return _ImpactAreaScenarioResults;
```

Comment:
we don't have the logic in place to abort if can't compute

Reply:
refactoring to stop passing around null objects would be a solid reduction in technical debt